### PR TITLE
Logging enhancements

### PR DESCRIPTION
**Pull request by alexluckett**

### CODE REVIEW

File: src/plugins/auth.js [hunk 1]
```diff
@@ -38,27 +38,37 @@ export const auth = {
           const user = artifacts.decoded.payload
 
           if (!user) {
+            logger.error("Couldn't validate auth. User is missing.")
```
Comment (colinrotherham) [inline src/plugins/auth.js:41]: We say "authorisation" in full below versus "auth" here

Shall we flip the negative and keep it simple?

Do we always put full stops in errors?

```suggestion
            logger.error('Authentication error: User missing')
```

File: src/plugins/auth.js [hunk 2]
```diff
@@ -38,27 +38,37 @@ export const auth = {
...
               isValid: false
             }
           }
 
-          const groups = Array.isArray(user.groups) ? user.groups : []
+          const { preferred_username: preferredUsername, groups = [] } = user
+
+          if (!preferredUsername) {
+            logger.error(
+              "Couldn't validate auth. preferred_username is missing."
```
Comment (colinrotherham) [inline src/plugins/auth.js:51]:
```suggestion
              'Authentication error: Missing preferred_username'
```

Comment (alexluckett) [reply src/plugins/auth.js:51]: Much nicer :)

File: src/plugins/auth.js [hunk 3]
```diff
@@ -38,27 +38,37 @@ export const auth = {
...
 
           logger.debug(
-            `User ${user.preferred_username}: validating against groups: ${groups.join(', ')}`
+            `User ${preferredUsername}: validating against groups: ${groups.length ? groups.join(', ') : '[]'}`
           )
 
           if (!groups.includes(roleEditorGroupId)) {
             logger.warn(
-              `User ${user.preferred_username}: failed authorisation. "${roleEditorGroupId}" not in groups.`
+              `User ${preferredUsername}: failed authorisation. "${roleEditorGroupId}" not in groups.`
```
Comment (colinrotherham) [inline src/plugins/auth.js:64]: Want to keep this full stop? Checked checked any others

```suggestion
              `User ${preferredUsername}: failed authorisation. "${roleEditorGroupId}" not in groups`
```

**Review by colinrotherham — APPROVED** [review]

File: src/plugins/auth.js (hunk 3 above)
Comment (alexluckett) [reply src/plugins/auth.js]: Removed a few

**Pull request merged**